File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-variance-swaps-5-4/VarianceAmount.cpp
// VarianceAmount.cpp 
#include "VarianceAmount.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

VarianceAmount::VarianceAmount(TiXmlNode* xmlNode)
: CalculatedAmount(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //varianceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* varianceNode = xmlNode->FirstChildElement("variance");

   if(varianceNode){varianceIsNull_ = false;}
   else{varianceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- varianceNode , address : " << varianceNode << std::endl;
   #endif
   if(varianceNode)
   {
      if(varianceNode->Attribute("href") || varianceNode->Attribute("id"))
      {
          if(varianceNode->Attribute("id"))
          {
             varianceIDRef_ = varianceNode->Attribute("id");
             variance_ = boost::shared_ptr<Variance>(new Variance(varianceNode));
             variance_->setID(varianceIDRef_);
             IDManager::instance().setID(varianceIDRef_,variance_);
          }
          else if(varianceNode->Attribute("href")) { varianceIDRef_ = varianceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { variance_ = boost::shared_ptr<Variance>(new Variance(varianceNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<Variance> VarianceAmount::getVariance()
{
   if(!this->varianceIsNull_){
        if(varianceIDRef_ != ""){
             return boost::shared_static_cast<Variance>(IDManager::instance().getID(varianceIDRef_));
        }else{
             return this->variance_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Variance>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-variance-swaps-5-4/VarianceAmount.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class VarianceAmount : CalculatedAmount
    {
        public VarianceAmount(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode varianceNode = xmlNode.SelectSingleNode("variance");
            
            if (varianceNode != null)
            {
                if (varianceNode.Attributes["href"] != null || varianceNode.Attributes["id"] != null) 
                {
                    if (varianceNode.Attributes["id"] != null) 
                    {
                        varianceIDRef_ = varianceNode.Attributes["id"].Value;
                        Variance ob = new Variance(varianceNode);
                        IDManager.SetID(varianceIDRef_, ob);
                    }
                    else if (varianceNode.Attributes["href"] != null)
                    {
                        varianceIDRef_ = varianceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        variance_ = new Variance(varianceNode);
                    }
                }
                else
                {
                    variance_ = new Variance(varianceNode);
                }
            }
            
        
        }
        
    
        #region Variance_
        private Variance variance_;
        public Variance Variance_
        {
            get
            {
                if (this.variance_ != null)
                {
                    return this.variance_; 
                }
                else if (this.varianceIDRef_ != null)
                {
                    variance_ = IDManager.getID(varianceIDRef_) as Variance;
                    return this.variance_; 
                }
                else
                {
                      return this.variance_; 
                }
            }
            set
            {
                if (this.variance_ != value)
                {
                    this.variance_ = value;
                }
            }
        }
        #endregion
        
        public string varianceIDRef_ { get; set; }
        
    
        
    
    }
    
}

